### PR TITLE
Add logic for clicking on tabs and filtering the parcel list to show on the tabs

### PR DESCRIPTION
**Pull request by Kennard123661**
Update tabs Ui, 
filters between delivered and undelivered, 
add gui test to ensure that clicking on the tabs work
system tests checks ui lists and logic lists to ensure that they are correct for every system operation (in sync) 
update parcel to ensure that they parcels with different status can be equals (so that users editing one parcel to the next status (completed) will not produce duplicates on parcel updates)

### CODE REVIEW

**Comment by Kennard123661** [comment]
image

These are event listener methods, so ignore Codacy's input on this.

For the fall in coverage, this is due to the lack of tests for TabCommand. TabCommand was written by @vicisapotato , she will be handling testing for her command to ensure she meets the end-to-end requirements for CS2103T

**Comment by Kennard123661** [comment]
image

This statement is needed to check reference equality, ensuring that they are referencing the same object (not state equality)

**Comment by Kennard123661** [comment]
image

Null Pointer exception used by tests (this is needed) Please ignore all of Codacy's checks.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -50,15 +55,35 @@ public ModelManager(ReadOnlyAddressBook addressBook, UserPrefs userPrefs) {
...
 
     public ModelManager() {
         this(new AddressBook(), new UserPrefs());
     }
 
+    /**
+     * Updates Delivered and UndeliveredParcelList and resets the Active List to the correct reference
+     */
+    private void updatedDeliveredAndUndeliveredList() {
+        boolean isActiveDelivered = activeFilteredList == filteredDeliveredParcels;
```
Comment (ktnc985) [inline src/main/java/seedu/address/model/ModelManager.java]: Add a comment to to say that you are checking for reference equality so that people will know that what you did was intended

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 2]
```diff
@@ -58,10 +57,23 @@
...
+    public static final String VALID_STATUS_DELIVERING = "DELIVERING";
+    public static final String VALID_STATUS_OVERDUE = "OVERDUE";
+    public static final String VALID_STATUS_COMPLETED = "COMPLETED";
+    public static final String STATUS_DESC_PENDING = " " + PREFIX_STATUS + VALID_STATUS_PENDING;
+    public static final String STATUS_DESC_DELIVERING = " " + PREFIX_STATUS + VALID_STATUS_DELIVERING;
+    public static final String STATUS_DESC_COMPLETED = " " + PREFIX_STATUS + VALID_STATUS_COMPLETED;
+
+    // list of tags
+    public static final String VALID_TAG_FROZEN = "FROZEN";
+    public static final String VALID_TAG_FRAGILE = "FRAGILE";
```
Comment (ktnc985) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:72]: Changing of tags should have been done as a separate PR, but it's fine this time

Comment (Kennard123661) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java:72]: yeap I agree, but I wanted to make testing easier for everyone, cuz now our test codes are all over the place. so I wanted to standardize it so easier for everyone to test, rather than randomly writing some numbers

**Review by ktnc985 — APPROVED** [review]
